File: Data Structures & Algorithms/maximum-product-subarray/submission-1.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int res = nums[0];
        int currMax = 1;
        int currMin = 1;

        for (int num : nums) {
            int temp = currMax * num;
            currMax = max(max(num * currMax, num * currMin), num);
            currMin = min(min(temp, num * currMin), num);
            res = max(res, currMax);
        }
        return res;
    }

};

File: Data Structures & Algorithms/arranging-coins/submission-1.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
        l, r = 0, math.ceil(n / 2)
        res = 0

        while l <= r:
            m = (l + r) // 2

            total = m * (m + 1) // 2

            if total <= n:
                l = m + 1
                res = max(res, m)
            else:
                r = m - 1
            
        return res